src/widget/wnumberpos.h

```diff
@@ -20,26 +20,17 @@ class WNumberPos : public WNumber {
 
   private slots:
     void setValue(double dValue) override;
-    void slotSetPosition(double);
-    void slotSetTrackSampleRate(double dSampleRate);
-    void slotSetTrackSamples(double dSamples);
+    void slotSetTimeElapsed(double);
     void slotSetDisplayMode(double);
 
   private:
 
     TrackTime::DisplayMode m_displayMode;
 
-    // Old value set
-    double m_dOldPosition;
-    double m_dTrackSamples;
-    double m_dTrackSampleRate;
-    // True if remaining content is being shown
-    bool m_bRemain;
+    double m_dOldTime;
+    ControlProxy* m_pTimeElapsed;
+    ControlProxy* m_pTimeRemaining;
     ControlProxy* m_pShowTrackTimeRemaining;
-    // Pointer to control object for position, rate, and track info
-    ControlProxy* m_pVisualPlaypos;
```

Comment:
Did you check how many controller scripts are using this value? Do we still have a way to determine the relative position within a track and to jump to a relative position?

I remember that I used those values in my controller scripts for implementing a Quick Search function for loaded tracks: By pressing shift and rotating the jog wheel a single revolution lets you move through the whole track. Some kind of circular strip search, very convenient. Of course, the ability for reading and updating the relative play position withinin a track would even be more useful!

But I think we need to take backward compatibility into account, maybe marking those controls as deprecated and removing them in one of the following version.

Reply:
This isn't removing the ControlObject, just a ControlProxy accessing it for the WNumberPos widget.

Reply:
In general, yes we need to be very careful about removing ControlObjects because of backwards compatibility. This also means we need to be careful choosing names for them that we won't regret. :)